packages/runtime/src/next/client.ts

```diff
@@ -175,35 +175,36 @@ export function componentDocumentToRootEmbeddedDocument({
   type: string
   hasFallback?: boolean
 }): EmbeddedDocument {
-  const { data, locale, id } = document
+  const { data: rootElement, locale, id } = document
 
-  if (data != null && data.type !== type) {
+  if (rootElement != null && rootElement.type !== type) {
     throw new Error(
-      `Type "${data.type}" does not match the expected type "${type}" from the snapshot`,
+      `Type "${rootElement.type}" does not match the expected type "${type}" from the snapshot`,
     )
   }
 
-  const rootElement = data ?? {
+  const initialRootElement = {
     // Fallback rootElement
     // Create a stable uuid so two different clients will have the same empty element data.
     // This is needed to make presence feature work for an element that is not yet created.
-    key: deterministicUUID({ id, locale, seed: document.seed }),
+    key:
+      rootElement != null
+        ? rootElement?.key
```

Comment:
Optional chaining is unnecessary here since we check `rootElement` is not `null`

Reply:
Good catch, I was trying to do a type deduction with the optional chaining at first, must've left it behind.